device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -199,10 +199,10 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
             const unsigned int nBlocks =
                 (n_in_params + nThreads - 1) / nThreads;
 
-            kernels::apply_interaction<std::decay_t<detector_type>>
-                <<<nBlocks, nThreads, 0, stream>>>(
-                    m_cfg, {det_view, n_in_params, in_params_buffer,
-                            param_liveness_buffer});
+            kernels::apply_interaction<std::decay_t<detector_type>>(
```

Comment:
Would it not be possible to come up with a design in which the called (templated) function could deduce its template parameters? 🤔 Unfortunately none of the provided parameters are `detector_type` directly. But maybe some formalism could still be found for such a setup.

It would just make the API of the function less error prone.

Reply:
Should be doable after #921.